workflow/scripts/plot-assembly-comparison.py

```diff
@@ -0,0 +1,154 @@
+import sys
+from altair.vegalite.v4.schema.channels import Opacity
+import pandas as pd
+import pysam
+import altair as alt
+
+sys.stderr = open(snakemake.log[0], "w")
+
+data = pd.DataFrame()
+
+
+def register_lengths(sample, file_list, state, amplicon_state, data):
+    for file, assembler in zip(file_list, snakemake.params.assembler):
+        if state in ("initial", "scaffolded"):
+            with pysam.FastxFile(file) as infile:
+                data = data.append(
+                    {
+                        "Sample": sample,
+                        "Assembler": assembler,
+                        "Amplicon": amplicon_state,
+                        "length (bp)": max(len(contig.sequence) for contig in infile),
+                        "State": state,
+                    },
+                    ignore_index=True,
+                )
+        else:
+            quastDf = pd.read_csv(file, sep="\t")
+            data = data.append(
+                {
+                    "Sample": sample,
+                    "Assembler": assembler,
+                    "Amplicon": amplicon_state,
+                    "length (bp)": quastDf.loc[0, "N50"],
+                    "State": "N50",
+                    "Genome fraction (%)": quastDf.loc[0, "Genome fraction (%)"]
+                    if "Genome fraction (%)" in quastDf.columns
+                    else float("nan"),
+                },
+                ignore_index=True,
+            )
+    return data
+
+
+for sample, amplicon_state in zip(
+    snakemake.params.samples, snakemake.params.amplicon_state
+):
+
+    def load(inputfile, label=None):
+        return register_lengths(
+            sample,
+            [x for x in snakemake.input.get(inputfile) if sample in x],
+            label or inputfile,
+            amplicon_state,
+            data,
+        )
+
+    data = load("initial")
+    data = load("final", "scaffolded")
+    data = load("quast", "N50")
+
+data.replace({"Amplicon": {0: "Shotgun", 1: "Amplicon"}}, inplace=True)
+data.replace(
+    {
+        "Assembler": {
+            "megahit-std": "MEGAHIT",
+            "megahit-meta-large": "MEGAHIT meta-large",
+            "megahit-meta-sensitive": "MEGAHIT meta-sensitive",
+            "trinity": "Trinty",
+            "velvet": "Velvet",
+            "metaspades": "SPAdes meta",
+            "spades": "SPAdes",
+            "coronaspades": "SPAdes corona",
+            "rnaviralspades": "SPAdes RNA viral",
+        }
+    },
+    inplace=True,
+)
+
+data.to_csv(snakemake.output[1])
+
+height = 200
+width = 50
+
+plot_bp = (
+    alt.Chart(data)
+    .encode(
+        y=alt.Y(
+            "length (bp)",
+            scale=alt.Scale(domain=[0, 35000], clamp=True),
+            axis=alt.Axis(tickCount=5),
+        ),
+        x=alt.X("State", title=None, sort=["initial", "scaffolded", "N50"]),
```

Comment:
We should rename the contig state in something more meaningful for outsiders and put the new 50 in the first second place, maybe like:
`N50, Largest Contig, Scaffold`

Reply:
I agree N50 first makes more sense, as it is related with the initial contigs